src/servers/kfserving_http_server.cc

```diff
@@ -0,0 +1,949 @@
+    // TODO Try to verify that model name is valid
+    LOG_VERBOSE(1) << "model_name: " << model_name << ", rest: " << rest;
+
+    // status
+    if (rest == "/metadata") {
+      HandleStatus(req, model_name);
+      return;
+    }
+    // health
+    if ((rest == "") && (model_name != "metadata")) {
+      HandleHealth(req, model_name);
+      return;
+    }
+    // infer
+    if (rest == ":predict") {
+      HandleInfer(req, model_name);
+      return;
+    }
+  }
+
+  LOG_VERBOSE(1) << "KFServing HTTP error: " << req->method << " "
+                 << req->uri->path->full << " - "
+                 << static_cast<int>(EVHTP_RES_BADREQ);
+  evhtp_send_reply(req, EVHTP_RES_BADREQ);
+}
+
+void
+KFServingHTTPAPIServer::HandleHealth(
+    evhtp_request_t* req, const std::string& model_name)
+{
+  if (req->method != htp_method_GET) {
+    evhtp_send_reply(req, EVHTP_RES_METHNALLOWED);
+    return;
+  }
+
+  if (model_name.empty()) {
+    evhtp_send_reply(req, EVHTP_RES_BADREQ);
+    return;
+  }
+
+  bool ready = true;
+  TRTSERVER_Protobuf* server_status_protobuf = nullptr;
+  TRTSERVER_Error* err = TRTSERVER_ServerModelStatus(
+      server_.get(), model_name.c_str(), &server_status_protobuf);
+  if (err == nullptr) {
+    const char* status_buffer;
+    size_t status_byte_size;
+    err = TRTSERVER_ProtobufSerialize(
+        server_status_protobuf, &status_buffer, &status_byte_size);
+    if (err == nullptr) {
+      ServerStatus server_status;
+      if (!server_status.ParseFromArray(status_buffer, status_byte_size)) {
+        err = TRTSERVER_ErrorNew(
+            TRTSERVER_ERROR_UNKNOWN, "failed to parse server status");
+        if (err == nullptr) {
+          const auto& itr = server_status.model_status().find(model_name);
+          if (itr == server_status.model_status().end()) {
+            err = Error(
+                RequestStatusCode::INTERNAL,
+                "unable to find health of \"" + model_name + "\"");
+          } else {
+            std::map<int64, ModelVersionStatus> model_versions =
+                itr->second.version_status();
+            for (auto mit = model_versions.begin(); mit != model_versions.end();
+                 ++mit) {
+              if (mit->ready_state() == ModelReadyState::MODEL_READY) {
+                ready = false;
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  TRTSERVER_ProtobufDelete(server_status_protobuf);
+  RequestStatus request_status;
+  RequestStatusUtil::Create(
+      &request_status, err, RequestStatusUtil::NextUniqueRequestId(),
+      server_id_);
+
+  evhtp_headers_add_header(
+      req->headers_out,
+      evhtp_header_new(
+          kStatusHTTPHeader, request_status.ShortDebugString().c_str(), 1, 1));
+
+  evhtp_send_reply(
+      req, (ready && (err == nullptr)) ? EVHTP_RES_OK : EVHTP_RES_BADREQ);
+
+  TRTSERVER_ErrorDelete(err);
+}
+
+void
+KFServingHTTPAPIServer::HandleStatus(
```

Comment:
I would not implement the "metadata" endpoint. Our status is not in a standard format and we don't want to create that standard format at this point.

Reply:
Removed